katdal/test/test_dataset.py

```diff
@@ -19,8 +19,64 @@
 from __future__ import print_function, division, absolute_import
 
 from nose.tools import assert_equal
+import numpy as np
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+from katpoint import Target, Antenna, Timestamp, rad2deg
 
-from katdal.dataset import _selection_to_list
+from katdal.dataset import (_selection_to_list, DataSet, Subarray,
+                            DEFAULT_VIRTUAL_SENSORS)
+from katdal.sensordata import SensorCache
+from katdal.categorical import CategoricalData
+from katdal.spectral_window import SpectralWindow
+
+
+class MinimalDataSet(DataSet):
+    """Minimal data set containing a single target track."""
+    def __init__(self, target, subarray, spectral_window, timestamps):
+        super(MinimalDataSet, self).__init__(name='test', ref_ant='array')
+        num_dumps = len(timestamps)
+        num_chans = spectral_window.num_chans
+        num_corrprods = len(subarray.corr_products)
+        dump_period = timestamps[1] - timestamps[0]
+
+        def constant_sensor(value):
+            return CategoricalData([value], [0, num_dumps])
+
+        self.subarrays = [subarray]
+        self.spectral_windows = [spectral_window]
+        sensors = {}
+        for ant in subarray.ants:
+            sensors['Antennas/%s/antenna' % (ant.name,)] = constant_sensor(ant)
+            az, el = target.azel(timestamps, ant)
+            sensors['Antennas/%s/az' % (ant.name,)] = az
+            sensors['Antennas/%s/el' % (ant.name,)] = el
+        array_ant_fields = ['array'] + ant.description.split(',')[1:5]
+        array_ant = Antenna(','.join(array_ant_fields))
```

Comment:
This bit of magic could use some comments (and maybe it's time for a utility function in katpoint).

Reply:
Agreed.